packages/react-cookie/src/hooks.ts

```diff
@@ -0,0 +1,52 @@
+import {useContext, useState} from 'react';
+import {CookieSerializeOptions} from 'cookie';
+import {CookieContext} from './context';
+
+const NO_MANAGER_ERROR = [
+  'Could not find react-cookie context.',
+  'You might be missing the <Cookie />',
+  'component in your application component tree.',
+].join(' ');
+
+export function useCookie(
+  key: string,
+): [
+  string | undefined,
+  (value?: string, options?: CookieSerializeOptions) => void
+] {
+  const manager = useContext(CookieContext);
+
+  if (!manager) {
+    throw new Error(NO_MANAGER_ERROR);
+  }
+
+  const [cookie, setCookieValue] = useState(
+    () => (manager ? manager.getCookie(key) : undefined),
+  );
+
+  const setCookie = (value?: string, options?: CookieSerializeOptions) => {
+    if (!value) {
+      setCookieValue('');
+      manager.removeCookie(key);
+
+      return;
+    }
+
+    setCookieValue(value);
+    manager.setCookie(key, {value, ...options});
+  };
+
+  return [cookie, setCookie];
```

Comment:
This would break our lint rule https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/react-hooks-strict-return.md. 

I opted for "setting" to null/undefined to remove the cookie, but I did not add documentation or tests for this feature. Those are both added now and I think provides a simple API for consumers. Do you think an explicit `removeCookie()` function is needed?

Reply:
Personally I like the consistency with koa of doing `set('name', null)` as long as it's documented

Reply:
If it makes sense to follow our rule, I guess it's fine.  I'm not 100% sure about how strict we about following it.

Is there any kind of type-hint we can provide like:
```
export const UNSET = '';
...
setCookie(name: string, value: string | CookieValue | UNSET)
```

Eww :/